src/cascadia/LocalTests_SettingsModel/SerializationTests.cpp

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "../TerminalSettingsModel/ColorScheme.h"
+#include "../TerminalSettingsModel/CascadiaSettings.h"
+#include "JsonTestClass.h"
+#include "TestUtils.h"
+#include <defaults.h>
+#include "../ut_app/TestDynamicProfileGenerator.h"
+
+using namespace Microsoft::Console;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+using namespace WEX::Common;
+using namespace winrt::Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::TerminalControl;
+
+namespace SettingsModelLocalTests
+{
+    // TODO:microsoft/terminal#3838:
+    // Unfortunately, these tests _WILL NOT_ work in our CI. We're waiting for
+    // an updated TAEF that will let us install framework packages when the test
+    // package is deployed. Until then, these tests won't deploy in CI.
+
+    class SerializationTests : public JsonTestClass
+    {
+        // Use a custom AppxManifest to ensure that we can activate winrt types
+        // from our test. This property will tell taef to manually use this as
+        // the AppxManifest for this test class.
+        // This does not yet work for anything XAML-y. See TabTests.cpp for more
+        // details on that.
+        BEGIN_TEST_CLASS(SerializationTests)
+            TEST_CLASS_PROPERTY(L"RunAs", L"UAP")
+            TEST_CLASS_PROPERTY(L"UAP:AppXManifest", L"TestHostAppXManifest.xml")
+        END_TEST_CLASS()
+
+        TEST_METHOD(GlobalSettings);
+        TEST_METHOD(Profile);
+        TEST_METHOD(ColorScheme);
+        TEST_METHOD(CascadiaSettings);
+
+        TEST_CLASS_SETUP(ClassSetup)
+        {
+            InitializeJsonReader();
+            return true;
+        }
+
+    private:
+        std::string toString(const Json::Value& json)
+        {
+            // set indentation to empty string to remove newlines
+            // enableYAMLCompatibility adds a space after ':'
+            Json::StreamWriterBuilder builder;

```

Comment:
Oof. Isn't this expensive to keep building this Builder on every call to the method? Shouldn't it be static or something? I know it's test code and we care less... but this seems a particularly hefty cost.